src/user.rs

```diff
@@ -0,0 +1,177 @@
+pub async fn post_user(url: &str, user_json: &str) -> Result<User, Box<dyn std::error::Error>> {
+    let post: User = serde_json::from_str(user_json)?;
+
+    post.validate()?; // validation requirements defined in struct definition
+
+    let client = reqwest::Client::new();
+    let resp = client.post(url).json(&post).send().await?;
+
+    let body = serde_json::from_str(resp.text().await?.as_str())?;
+
+    Ok(body)
+}
+
+/// Defining Structs used to serialize and deserialize responses from reqwest for a user
+#[derive(Serialize, Deserialize, validator_derive::Validate, Default, Debug)]
+pub struct User {
+    #[serde(default)]
+    pub id: usize,
+    #[serde(default)]
+    #[validate(length(min = 3))]
+    pub name: String,
+    #[serde(default)]
+    pub username: String,
+    #[serde(default)]
+    #[validate(email)]
+    pub email: String,
+    #[serde(default)]
+    pub address: Address,
+    #[serde(default)]
+    pub phone: String,
+    #[serde(default)]
+    pub website: String,
+    #[serde(default)]
+    pub company: Company,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct Address {
+    #[serde(default)]
+    pub street: String,
+    #[serde(default)]
+    pub suite: String,
+    #[serde(default)]
+    pub city: String,
+    #[serde(default)]
+    pub zipcode: String,
+    #[serde(default)]
+    pub geo: Geo,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct Geo {
+    #[serde(default)]
+    pub lat: String,
+    #[serde(default)]
+    pub lng: String,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct Company {
+    #[serde(default)]
+    pub name: String,
+    #[serde(rename = "catchPhrase", default)]
+    //snake case to camel for JSON format to keep idiomatic rust var names.
+    pub catch_phrase: String, // NOTE: JSON uses catchPhrase
+    #[serde(default)]
+    pub bs: String,
+}
+
+#[cfg(test)]
+mod tests {
+    //! Tests for `get_user` and `post_user` methods
+    //! Note: the endpoints for this lib are HARD CODED. only to be used with `https://jsonplaceholder.typicode.com/api/v1/users`
+
+    use super::*;
+
+    use serde_json::json;
+    use tokio;
+
+    const URL: &str = "https://jsonplaceholder.typicode.com/api/v1/users";
+
+    #[tokio::main]
+    #[test]
+    async fn it_gets_users() -> Result<(), Box<dyn std::error::Error>> {
+        let address_check = Address {
+            street: String::from("Douglas Extension"),
+            suite: String::from("Suite 847"),
+            city: String::from("McKenziehaven"),
+            zipcode: String::from("59590-4157"),
+            geo: Geo {
+                lat: String::from("-68.6102"),
+                lng: String::from("-47.0653"),
+            },
+        };
+
+        let users: Vec<User> = get_users(URL).await?;
+        // println!("---Response: (Address of a user):\n {:#?}", users[2].address.street);
+
+        assert_eq!(address_check.street, users[2].address.street);
```

Comment:
This address-based test is brittle.  I think you can verify that this succeeds by asserting that the response is OK and that the vector has a non-zero length.